File: python/candyland/game.py
from player import Player
from board import Board
from deck import Deck
import random
 
class Game(object):
	def __init__(self,count):
		
		if count > 4:
			return 
		self.winner = "None"
		self.player_count = count
		self.players = []

		game_pieces = ["Red","Blue","Green","Yellow"]
		for i in range (0, self.player_count):
			print "adding player"
			self.players.append(Player(random.choice(game_pieces)))
		self.board = Board()
		self.card_deck = Deck()

	def play_game(self):
		rounds = 0
		while (self.winner == 'None'):
			rounds = rounds + 1
			self.play_round()
		print "%s wins after %s" %(self.winner,rounds)
		return rounds

	def play_round(self):
		
		for player in self.players:
			self.play_turn(player)
			if player.position >= self.board.board_length -1:
				self.winner = player.name
				break
			

	def play_turn(self,player):
		
		if player.stuck == 0:
			card = self.card_deck.draw_card()
			self.move_token(player,card)
			print_position = player.position + 1
			pretty_card = card.to_string()
			pretty_space = self.board.get_space_from_position(player.position).color
			print "%s drew %s and moved to space %d which is %s" %(player.name,pretty_card,player.position,pretty_space)
		else:
			print "%s is stuck! Skipping turn" % player.name
			player.stuck = 0

	def move_token(self,player,card):
		stuck = 0
		for a in range (0,card.move_count):
			if card.color == "Pink":
				next_space = 0
			else:
				next_space = player.position + 1
			current_position = player.position

			for i in range(next_space,self.board.board_length):
				space = self.board.get_space_from_position(i)
				#print "Trying %s %s for %s %s" %(space.color,space.label,card.color,card.label)
				if space.color == card.color or space.color == "All":
					if space.color == "Pink" and space.label == card.label:
						spaces = i
						current_position = i
						break
					elif space.label == "Rainbow Trail":
						current_position = i + 55
						break
					elif space.label == "Gumdrop Pass":
						current_position = i + 14
						break
					elif space.color != "Pink":
						current_position = i
						if space.label == "Sticky":
							stuck = 1
						break
			player.position = current_position
			player.stuck = stuck 

File: java/candyland/Game.java
package candyland;
import candyland.*;
import java.util.*;

public class Game {
	List<Player> players;
	Board board;
	Deck deck;
	String winner;
	int player_count;
	public Game(int player_count) {
		this.player_count = player_count;
		this.winner = "None";

		players = new ArrayList<Player>();
		String[] game_peices = {"Red","Blue","Green","Yellow"};
		for (int i = 0; i < this.player_count;i++) {
			Player p = new Player(game_peices[i]);
			this.players.add(p);
		}

		this.board = new Board();
		this.deck = new Deck();

	}

	public int play_game() {
		int rounds = 0;
		while (this.winner.equals("None")) {
			rounds++;
			play_round();
		}
		System.out.println(this.winner + " wins after " + rounds + " rounds!");
		return rounds;

	}

	private void play_round() {
		for (Player p : this.players) {
			play_turn(p);
			if (p.position >= (this.board.board_length - 1)) {
				this.winner = p.name;
				break;
			}
		}

	}

	private void play_turn(Player p) {
		//System.out.println(p.stuck);
		if (!p.stuck) {
			Card card = this.deck.draw_card();
			move_token(p,card);
			int print_postion = p.position + 1;
			//System.out.println(print_postion);

			System.out.println(p.name + " drew " + card.to_string() + " and moves to space " + print_postion + " which is " + this.board.get_space_from_positon(p.position).color);
		}
		else {
			System.out.println(p.name + " is stuck! Skipping turn!");
			p.stuck = false;
		}


	}

	private void move_token(Player p, Card card) {
		boolean stuck = false;
		int current_postion = p.position;
		for (int m = 0; m < card.move_count;m++) {
			int next_space;

			if (card.color.equals("Pink")) {
				next_space = 0;
			}
			else if (current_postion == 0) {
				 next_space = 0;
			}
			else {
				next_space = current_postion + 1;
			}


			for (int i = next_space; i < this.board.board_length; i++) {
				Space current_space = this.board.get_space_from_positon(i);
				if (current_space.color.equals(card.color) || current_space.color.equals("All")) {
					if (current_space.color.equals("Pink") && current_space.label.equals(card.label)) {
						current_postion = i;
						break;
					}
					else if (current_space.label != null && current_space.label.equals("Rainbow Trail")) {
						current_postion = i + 55;
						break;
					}
					else if (current_space.label != null && current_space.label.equals("Gumdrop Pass")) {
						current_postion = i + 14;
					}
					else if (!(current_space.color.equals("Pink"))) {
						current_postion = i;
						if (current_space.label != null && current_space.label.equals("Sticky")) {
							stuck = true;
						}
						break;
					}
				}
			}
			p.position = current_postion;
			//System.out.println("Changing stuck to " + p.stuck);
			p.stuck = stuck;
		}

	}
}